tests/ui/impl-trait/method-resolution2.rs

```diff
@@ -0,0 +1,24 @@
+trait Trait {}
+
+impl Trait for i32 {}
+
+struct Bar<T>(T);
+
+impl Bar<u32> {
+    fn foo(self) {}
+}
+
+impl<T: Trait> Bar<T> {
+    fn foo(self) {}
+}
+
+fn foo(x: bool) -> Bar<impl Trait> {
+    //~^ ERROR type annotations needed
+    if x {
+        let x = foo(false);
+        x.foo();
+    }
+    todo!()
```

Comment:
maybe instead return `Bar<i32>` here and change this to `check-pass`? I think this test is intended to show that we still reject the `Bar<u32>` impl because `u32: Trait` does not hold. So method selection succeeds and the ambiguity only happens after that